cli/azd/pkg/commands/builder.go

```diff
@@ -25,28 +29,57 @@ func RegisterDependenciesInCtx(
 	cmd *cobra.Command,
 	rootOptions *internal.GlobalCommandOptions,
 ) (context.Context, error) {
+
 	// Set the global options in the go context
 	ctx = internal.WithCommandOptions(ctx, *rootOptions)
 	ctx = tools.WithInstalledCheckCache(ctx)
 
 	runner := exec.NewCommandRunner(cmd.InOrStdin(), cmd.OutOrStdout(), cmd.ErrOrStderr())
 	ctx = exec.WithCommandRunner(ctx, runner)
 
-	// Set default credentials used for operations against azure data/control planes
-	credentials, err := azidentity.NewAzureCLICredential(nil)
+	writer := cmd.OutOrStdout()
+
+	if os.Getenv("NO_COLOR") != "" {
+		writer = colorable.NewNonColorable(writer)
+	}
+
+	// To support color on windows platforms which don't natively support rendering ANSI codes
+	// we use colorable.NewColorableStdout() which creates a stream that uses the Win32 APIs to
+	// change colors as it interprets the ANSI escape codes in the string it is writing.
+	if writer == os.Stdout {
+		writer = colorable.NewColorableStdout()
+	}
+
+	authManager, err := auth.NewManager(writer, config.NewManager())
 	if err != nil {
-		panic("failed creating azure cli credential")
+		return ctx, fmt.Errorf("creating auth manager: %w", err)
+	}
+
+	var credential azcore.TokenCredential
+
+	// TODO(ellismg): This is a hack so that we don't fail for `login` when we construct the root context if a user
+	// is not logged in. This is super fragile, but we should be able to clean it up soon with Wei's work.
+	if cmd.Use != "login" {
+		_, cred, _, err := authManager.GetSignedInUser(ctx)
+		if err != nil {
+			return ctx, fmt.Errorf("fetching current user: %w", err)
+		}
+		credential = cred
+	} else {
+		credential = &panicCredential{}
 	}
-	ctx = identity.WithCredentials(ctx, credentials)
 
 	azCliArgs := azcli.NewAzCliArgs{
 		EnableDebug:     rootOptions.EnableDebugLogging,
 		EnableTelemetry: rootOptions.EnableTelemetry,
 		CommandRunner:   runner,
 	}
 
+	// Set default credentials used for operations against azure data/control planes
+	ctx = identity.WithCredentials(ctx, credential)
```

Comment:
Curious if we can combine the `identity` & `auth` packages.  The identity package today only has the `WithCredentials` and `GetCredentials` context helpers.

Reply:
Sounds good to me.  I sort of prefer the `auth` package name here over `identity`, but I could be fine changing things.  I think that over the next few weeks we will remove `WithCredentials` and `GetCredentials` as we move away from using the credential object as our IoC system and so at that point we could just remove `identity`.

Reply:
I took a few runs at trying to remove the `identity` package but it was starting to blow the size of this PR up even more.  I did some cleanup in this area instead and will remove the `identity` package in a future PR.